### ISSUE TITLE
Cosine loss negative

### ISSUE THREAD

**Issue by anushl9o5** [wangruixing/DeepUPE#45]
Hi, 
On what axis to do l2 norm and on what axis to use cosine distance from tf ?
Thanks!

**Comment by wangruixing** [comment]
img_ref = tf.nn.l2_normalize(train_samples['image_gt'],3)
ref_p = tf.nn.l2_normalize(predict,3)
loss_cos = tf.losses.cosine_distance(img_ref, ref_p,dim=3)

With this loss function, when I retrained the network around iter 11000+, the PSNR has been increased 0.8. If you have further implementation problem, you can contact me by email: rxwang@cse.cuhk.edu.hk   :)

**Comment by anushl9o5** [comment]
Here 'ref_p' is the result of dividing the input image by the illuminance map, right? 
I have sent a few more doubts to you email. 
Thanks for the help!
